docs/topics/native/native-c-interop-stability.md

```diff
@@ -0,0 +1,180 @@
+
+> This document describes not only problems that do happen often in practice, but also problems that might happen
+> (but usually don't), and hypothetical potential problems as well.
+>
+{style="note"}
+
+There is an important distinction:
+- On each platform, Kotlin provides access to "system" native libraries through so-called
+  ["platform libraries"](native-platform-libs.md)
+- All other native libraries require additional configuration to be used from Kotlin, and this document calls them
+  "third-party libraries".
+
+These two kinds of native libraries have different compatibility properties.
+
+## Platform libraries
+
+Platform libraries are shipped with the Kotlin/Native compiler.
+So, using different versions of Kotlin in the project results in getting different versions of platform libraries.
+For Apple targets (like iOS), platform libraries are generated based on an Xcode version this version of the compiler
+supports.
+
+Native APIs of libraries shipped with Xcode SDKs change with an Xcode version.
+Even when such changes are source- and binary-compatible within the native language,
+they might become breaking for Kotlin due to the way the interoperability is implemented.
+
+As a result, updating the Kotlin version in a project might bring a breaking change in platform libraries.
+This might matter in two cases:
+- There is a source breaking change in platform libraries that affects the compilation of the source code in your
+  project.
+  Usually it should be trivial to fix.
+- There is a binary breaking change in platform libraries that affects some of your dependencies.
+  There is typically no easy workaround, and you will need to wait until the library developer will fix this on their
+  side, e.g., by updating the Kotlin version.
+
+The JetBrains team, whenever updates the Xcode version used to generate the platform libraries, makes a reasonable
+effort to avoid breaking changes in platform libraries: whenever a breaking change might occur, the team conducts impact
+analysis and either decides to ignore this particular change (because the affected API is not commonly used) or applies
+an ad hoc fix for this problem.
+
+Another potential reason for breaking changes in platform libraries is changes in the algorithm that translates
+native APIs to Kotlin.
+Just like above, the JetBrains team makes reasonable efforts to avoid breaking changes in such cases as well.
+
+## Importing third-party native libraries when the Xcode version doesn't match
+
+Apart from system libraries, Kotlin/Native allows importing third-party native libraries.
+For example,
+by using [CocoaPods integration](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-cocoapods-dsl-reference.html)
+or defining a [cinterops configuration](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-dsl-reference.html#cinterops).
+This has implications for the compatibility of your project with different Xcode versions.
```

Comment:
One aspect that might not be entirely clear to someone not familiar with C/Objective-C and the K/N integration with Xcode is: where exactly does Xcode come into play? Maybe it's worth pointing out that for Apple targets, when we run `.h` -> `.klib` translation, the `.h` files are resolved against the selected Xcode sysroot for this particular target, rather than a sysroot that we vendor, as is the case for other targets. That's the reason Xcode update affects your cinterops and also the reason why cross-cinterop from non-Apple hosts to Apple targets is not currently supported.